tests/functional/WebIfcApi.spec.ts

```diff
@@ -1,37 +1,454 @@
+        let vertexArray: Float32Array = ifcApi.GetVertexArray(geometry.GetVertexData(), geometry.GetVertexDataSize());
+        expect(vertexArray).toBeInstanceOf(Float32Array);
+    })
+    test('can return vertex array from a flat mesh', () => {
+        let geometryExpressId = geometries.get(1).expressID;
+        let flatMesh = ifcApi.GetFlatMesh(modelID, geometryExpressId);
+        let flatMesheGeometries: IfcGeometry = ifcApi.GetGeometry(modelID, flatMesh.geometries.get(0).geometryExpressID);
+        let geometryVertexArray: Float32Array = ifcApi.GetVertexArray(flatMesheGeometries.GetVertexData(), flatMesheGeometries.GetVertexDataSize());
+        expect(geometryVertexArray.length > 0).toBeTruthy();
+    })
+    test('can return geometry index datas from a flat mesh', () => {
+        let geometryExpressId = geometries.get(1).expressID;
+        let flatMesh = ifcApi.GetFlatMesh(modelID, geometryExpressId);
+        let flatMesheGeometries: IfcGeometry = ifcApi.GetGeometry(modelID, flatMesh.geometries.get(0).geometryExpressID);
+        let geometryIndexData: Uint32Array = ifcApi.GetIndexArray(flatMesheGeometries.GetIndexData(), flatMesheGeometries.GetIndexDataSize());
+        expect(geometryIndexData.length > 0).toBeTruthy();
+    })
+    test('expect correct vertex and index array from A flat Mesh', () => {
+        let flatMesh = ifcApi.GetFlatMesh(modelID, geometries.get(expectedVertexAndIndexDatas.geometryIndex).expressID);
+        let geometry = ifcApi.GetGeometry(modelID, flatMesh.geometries.get(0).geometryExpressID);
+        let geometryVertexArray = ifcApi.GetVertexArray(geometry.GetVertexData(), geometry.GetVertexDataSize());
+        let geometryIndexData = ifcApi.GetIndexArray(geometry.GetIndexData(), geometry.GetIndexDataSize());
+        let geometryIndexDatasString = geometryIndexData.join(",");
+        let geometryVertexArrayString = geometryVertexArray.join(",");
+        expect(geometryIndexDatasString).toEqual(expectedVertexAndIndexDatas.indexDatas);
+        expect(geometryVertexArrayString).toEqual(expectedVertexAndIndexDatas.vertexDatas);
+
+    })
+    test('can ensure the corret number of all streamed meshes ', () => {
+        let count: number = 0;
+        ifcApi.StreamAllMeshes(modelID, (mesh) => {
+            count++;
+        });
+        expect(count).toEqual(meshesCount);
+    })
+    test('can ensure the corret number of all streamed meshes with a given Types', () => {
+        let count: number = 0;
+        ifcApi.StreamAllMeshesWithTypes(modelID, [WebIFC.IFCEXTRUDEDAREASOLID], (mesh) => {
+            ++count
+        });
+        expect(count).toEqual(IFCEXTRUDEDAREASOLIDMeshesCount);
+    })
+});
+describe('WebIfcApi geometry transformation', () => {
+
+    test('should throw Error with message \'invalid matrix size\' when matrix size != 16', () => {
+        let f = ifcApi.SetGeometryTransformation(modelID, [1]);
+        expect(f).toThrow("invalid matrix size");
```

Comment:
something wrong here the correct code is the following one .
``` js 
        expect(() => {
            ifcApi.SetGeometryTransformation(modelID, [1]);
        }).toThrow("invalid matrix size");
```
this test will pass with this PR https://github.com/IFCjs/web-ifc/pull/234